src/test/scala/is/hail/expr/ir/CompileSuite.scala

```diff
@@ -528,4 +528,44 @@ class CompileSuite {
     assert(expected.length == actual.length)
     assert(actual.sameElements(expected))
   }
+
+  @Test
+  def testShortCircuitBooleans() {
+    val tin = TArray(TBoolean())
+    val tout = TTuple(TBoolean())
+    val irAnd = MakeTuple(Seq(ApplyBinaryPrimOp(DoubleAmpersand(), ArrayRef(In(0, tin), I32(0)), ArrayRef(In(0, tin), I32(1)))))
+    val irOr = MakeTuple(Seq(ApplyBinaryPrimOp(DoublePipe(), ArrayRef(In(0, tin), I32(0)), ArrayRef(In(0, tin), I32(1)))))
+
+    val fb = FunctionBuilder.functionBuilder[Region, Long, Boolean, Long]
+    doit(irAnd, fb)
+    val fAnd = fb.result()()
+
+    val fb2 = FunctionBuilder.functionBuilder[Region, Long, Boolean, Long]
+    doit(irOr, fb2)
+    val fOr = fb2.result()()
+
+    val region = Region()
+    val rvb = new RegionValueBuilder(region)
+
+    for (l: Option[Boolean] <- Seq(Some(true), Some(false), None); r: Option[Boolean] <- Seq(Some(true), Some(false), None)) {
```

Comment:
I prefer this on two lines, also if you use `{` and `}` and separate lines you can elide the `;`.